Makefile

```diff
@@ -54,7 +55,29 @@ test: clean-cov fmt vet $(GINKGO)
 ## install: Build and install kuadrantctl binary ($GOBIN or GOPATH/bin)
 .PHONY : install
 install: fmt vet
-	GOBIN=$(PROJECT_PATH)/bin $(GO) install
+	@set -e; \
+	GIT_SHA=$$(git rev-parse --short=7 HEAD 2>/dev/null) || { \
+		GIT_HASH=$${GITHUB_SHA:-NO_SHA}; \
+		IS_DIRTY=false; \
+	}; \
+	if [ -z "$$GIT_HASH" ]; then \
+		GIT_DIRTY=$$(git diff --stat); \
+		if [ -n "$$GIT_DIRTY" ]; then \
+			GIT_HASH=$${GIT_SHA}-dirty; \
+			IS_DIRTY=true; \
+		else \
+			GIT_HASH=$${GIT_SHA}; \
+			IS_DIRTY=false; \
+		fi; \
+	fi; \
+	LDFLAGS="-X 'github.com/kuadrant/kuadrantctl/version.GitHash=$$GIT_HASH'"; \
+	if [ "$$IS_DIRTY" = true ]; then \
+		LDFLAGS="$$LDFLAGS -X 'github.com/kuadrant/kuadrantctl/version.GitDirty=true'"; \
```

Comment:
`GitDirty` does not exist, does it?

Reply:
You're right - I'll amend that. Good spot! :D